pkgs/development/compilers/ghc/8.2.1.nix

```diff
@@ -32,33 +61,111 @@ in stdenv.mkDerivation (rec {
 
   postPatch = "patchShebangs .";
 
-  preConfigure = commonPreConfigure;
+  #v p dyn
+  preConfigure = stdenv.lib.optionalString (targetPlatform != hostPlatform)''
+    sed 's|#BuildFlavour  = quick-cross|BuildFlavour  = perf-cross|' mk/build.mk.sample > mk/build.mk
```

Comment:
This whole section of writing to `mk/build.mk` can be done with just one I/O operation using sed just as a stream editor, and the HEREDOC feature of bash. I know we all have fast I/O devices, but it would make the code shorter too.